GCS/nullprobe/nullprobe.go

```diff
@@ -0,0 +1,143 @@
+package hermes.nullprobe
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"io"
+	"os"
+	"time"
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/probes/options"
+	"github.com/google/cloudprober/targets/endpoint"
+	"github.com/googleinterns/step224-2020/blob/evan/hermesMain/cmd"
+)
+
+// HermesProbe holds aggregate information about all probe runs, per-target
+type HermesProbe struct {
+	name    string
+	conf       *ProbeConf
+	targets []string
+	opts    *options.Options
+
+	results map[string]*metrics.EventMetrics  // Results by target
+	l   *logger.Logger
+}
+
+// Init initializes the probe with the given params.
+func (p *HermesProbe) Init(name string, opts *options.Options) error {
+	p.conf, ok := opts.ProbeConf.(*ProbeConf)
+	if !ok {
+		return fmt.Errorf("Invalid argument: type of opts.ProbeConf does not match *ProbeConf")
```

Comment:
To make your testing simpler, you could make a little Enum of error types that you include in your errors. This way in your tests you could just compare error types as well as include the enum for your error in any relevant probe result/error/success metrics.

Totally optional though, you could also just open an issue to make this a future feature.

Reply:
I would only recommend making the errors more elaborate if the caller has the handle different errors in different ways.

If that's the case, then use custom error types that implement the error interface (instead of enums).
See https://blog.golang.org/go1.13-errors.